File: algorithms/implementation/utopian-tree/utopian-tree-java/utopian-tree-java/src/TesteUtopianTree.java
import org.junit.Test;

public class TesteUtopianTree {

    @Test
    public void testeCaso1() {

        int n = 0;
        int esperado = 1;

        int resultado = Resultado.utopianTree(n);

        assert resultado == esperado;
    }

    @Test
    public void testeCaso2() {

        int n = 1;
        int esperado = 2;

        int resultado = Resultado.utopianTree(n);

        assert resultado == esperado;
    }

    @Test
    public void testeCaso3() {

        int n = 4;
        int esperado = 7;

        int resultado = Resultado.utopianTree(n);

        assert resultado == esperado;
    }
}

File: algorithms/implementation/utopian-tree/utopian-tree-javascript/js/teste_utopianTree.js

import {utopianTree} from '../js/utopianTree.js';
import assert from 'assert';

describe('utopianTree', function() {

    it('A altura da árvore após um determinado número de ciclos é 1', function() {

        let n = 0;
        let esperado = 1;

        let resultado = utopianTree(n)

        assert.strictEqual(resultado, esperado)
    });
});

describe('utopianTree', function() {

    it('A altura da árvore após um determinado número de ciclos é 2', function() {

        let n = 1;
        let esperado = 2;

        let resultado = utopianTree(n)

        assert.strictEqual(resultado, esperado)
    });
});

describe('utopianTree', function() {

    it('A altura da árvore após um determinado número de ciclos é 7', function() {

        let n = 4;
        let esperado = 7;

        let resultado = utopianTree(n)

        assert.strictEqual(resultado, esperado)
    });
});